mula/tests/integration/test_api.py

```diff
@@ -151,19 +151,37 @@ def test_push_queue_full(self):
         self.scheduler.queue.maxsize = 1
 
         # Add one task to the queue
-        first_item = create_p_item(self.organisation.id, 0)
+        first_item = create_p_item(self.organisation.id, 1)
         response = self.client.post(
             f"/queues/{self.scheduler.scheduler_id}/push", json=json.loads(first_item.model_dump_json())
         )
         self.assertEqual(response.status_code, 201)
         self.assertEqual(1, self.scheduler.queue.qsize())
 
         # Try to add another task to the queue through the api
-        second_item = create_p_item(self.organisation.id, 1)
+        second_item = create_p_item(self.organisation.id, 2)
         response = self.client.post(f"/queues/{self.scheduler.scheduler_id}/push", json=json.loads(second_item.json()))
         self.assertEqual(response.status_code, 429)
         self.assertEqual(1, self.scheduler.queue.qsize())
 
+    def test_push_queue_full_high_priority(self):
+        # Set maxsize of the queue to 1
+        self.scheduler.queue.maxsize = 1
+
+        # Add one task to the queue
+        first_item = create_p_item(self.organisation.id, 1)
+        response = self.client.post(
+            f"/queues/{self.scheduler.scheduler_id}/push", json=json.loads(first_item.model_dump_json())
```

Comment:
The serialization and deserialization process doesn't make sense, either:
- `json=first_item.model_dump()` or
- `data=first_item.model_dump_json()`